src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -67,14 +68,26 @@ class Cargo(
         command: String,
         additionalArguments: List<String> = emptyList(),
         environmentVariables: Map<String, String> = emptyMap()
-    ): GeneralCommandLine =
-        GeneralCommandLine(pathToCargoExecutable)
+    ): GeneralCommandLine {
+        var args = additionalArguments
+        val cmdLine = GeneralCommandLine(pathToCargoExecutable)
             .withWorkDirectory(projectDirectory)
             .withParameters(command)
-            .withParameters(additionalArguments)
             .withEnvironment(CargoConstants.RUSTC_ENV_VAR, pathToRustExecutable)
             .withEnvironment(environmentVariables)
 
+        // Make output colored
+        if ((SystemInfo.isLinux || SystemInfo.isMac)
+            && command != "fmt"
```

Comment:
I think we should use a whitelist of commands which accept `--color` rather than a blacklist of those who don't, because you can add arbitrary subcommands to cargo, and they might not like `--color`, just like `cargo fmt`.